### DISCUSSION TITLE
Running a file on its own

### DISCUSSION THREAD

**Discussion by kenaryn** [mattwparas/steel#159] [Q&A] [answered]
Hi!

I have a newbie question: please how can I run a file written in Steel on its own?
Is there something like `steel myfile` to execute?

I even know not the extension of a Steel file and found no thing related to at https://mattwparas.github.io/steel/book/start/embedded.html

Thank you!

**Comment by mattwparas** [comment]
You should be able to just pass a file to the interpreter, assuming it is installed - also the extension is `.scm`:

```
steel my-file.scm
```

**Comment by kenaryn** [comment]
I want not to be a burden but I just cloned your repository, cd into it and ran `cargo run`. Where can I find the executable to be added to my `PATH` env var? I tried `fd -t x --full-path '/home/aurele/dev/steel'` but it returned no thing.

Also, what does `scm` stand for?

**Reply by mattwparas** [reply]
If you're running this way, just run `cargo run -- myfile.scm`

`scm` is one of many extensions for scheme, just happens to be the one I prefer

**Reply by kenaryn** [reply]
I am interested by the way you run it with `steel` command? How do you proceed please? As relying on a package manager from another stack seems clumsy to me.

**Reply by mattwparas** [reply]
You can install it to your system by running from the root of the repo:

```
cargo install --path .
```

**Reply by mattwparas** [reply, marked as answer]
You can also run `cargo build --release` and find the built executable in `target/release/` - and move it to wherever you see fit that is on your path

**Reply by kenaryn** [reply]
Alternatively to `cargo install --path` you mean? or after having run the previous command?
I know that is very beginner level question but I need to be sure to learn and grow.

**_Nota bene_**: it is indeed off-topic so I will attempt to gather the answer by myself.
The answer to my original question has been provided and I am grateful.

**Reply by mattwparas** [reply]
No worries - `cargo install --path` is just the cargo way of installing the executable to your path. I forget the exact location, but its the canonical way to install an executable from source with Rust. 

Running `cargo build --release` and then moving the executable to wherever you want it is just another way of installing it.
